### PR TITLE
Add default tests when creating an extension with Maven Mojo

### PR DESCRIPTION
**Pull request by danielpetisme**
Here is a proposal for #3585 

The only basic task we could do is ensure the extension is properly loaded (by scanning the startup output or programatically.

WDYT?

@gwenneg taked also to test hot reloading, I'm not sure to understand the goal.

### CODE REVIEW

**Comment by gwenneg** [comment]
You'll find more info about hot reload tests here: https://quarkus.io/guides/extension-authors-guide#testing-hot-reload

This was introducer in Quarkus `0.21.0` and should be used in all extensions.

**Comment by danielpetisme** [comment]
Thanks for the quick reply. I missed that point.

Open question, can we consider that the majority of the extensions will require HTTP and generate an opinionated  `Processor` which produce a Servlet + the corresponding `ProcessorTest` which will start the app in dev mode, verify the HTTP endpoint and verify hot reload.

It implies an opinionated vision but have the benefits to drive new extension development. That means **undertow-extension** and **Restassured** added by default in all the created extension.

Thoughts?

**Comment by gsmet** [comment]
I don't know, I wouldn't clutter the template too much, tbh.

**Comment by danielpetisme** [comment]
Me too...
I would like to provide a consistent experience between Quarkus app and extension generation:
- When generating an extension I should be able to run a smoke test right away

-  Enforce best practices (Like testing hot reloading for instance)

- Be self-documented/intuitive, the `FeatureBuildItem` is easy to understand and a simple test would be also helpful. 

I admit these objectives are to simple, there is a trade off to find.

Shall we only limit the current PR to the structure creation as is and complete the generation later on with more developer feedbacks?

File: devtools/maven/src/test/resources/expected/create-extension-pom-minimal/minimal-extension/deployment/src/test/java/org/acme/my/project/minimal/extension/deployment/MinimalExtensionProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+class MinimalExtensionProcessorTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+        .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class));
```
Comment (cescoffier) [inline devtools/maven/src/test/resources/expected/create-extension-pom-minimal/minimal-extension/deployment/src/test/java/org/acme/my/project/minimal/extension/deployment/MinimalExtensionProcessorTest.java]: Can you add comments to explain what the user should do in this code?

**Review by cescoffier — CHANGES_REQUESTED** [review]
The templates should be kept minimal. So they are good right now.
However, generating comments so the user knows where hacking is required may significantly improve the experience.

**Comment by cescoffier** [comment]
@danielpetisme also, you would need to rebase your PR.

**Comment by danielpetisme** [comment]
WDYT of [these comments].(https://github.com/quarkusio/quarkus/blob/3e08ec268c23bbe68555aaed252c0462c2b9672b/devtools/maven/src/main/resources/create-extension-templates/ProcessorTest.java)

**Comment by danielpetisme** [comment]
@cescoffier rebase done.

**Comment by machi1990** [comment]
@danielpetisme hi, we have tests failing, since the Azure artifacts expired I cannot tell if it is related or not. Do you mind having a look?

**Comment by cescoffier** [comment]
@danielpetisme ping? ^^^

**Comment by danielpetisme** [comment]
Looks like @machi1990 has been faster cf. #5438
Feel free to close it if it makes sense.
Sorry for the delay

**Comment by machi1990** [comment]
Ah @danielpetisme I did not see this initial PR at all, otherwise I would not have opened my PR. Sorry about that. You feedbacks/comments are however welcomed on the PR #5438 I opened.

**Comment by danielpetisme** [comment]
No problem. I prefer to see the issue to be tackled !!!

**Comment by machi1990** [comment]
Superseded by #5438

**Pull request closed without merging**